docs/proposals/scheduling/20231129-pod-migration-job-external-conditions.md

```diff
@@ -0,0 +1,142 @@
+---
+title: PodMigrationJob Support Eviction Conditions
+authors:
+  - "@sjtufl"
+reviewers:
+  - "@saintube"
+  - "@eahydra"
+  - "@zwzhang0107"
+  - "@hormes"
+creation-date: 2023-11-29
+last-updated: 2023-12-01
+---
+
+# PodMigrationJob Support Eviction Conditions
+
+## Table of Contents
+
+- [PodMigrationJob Support Eviction Conditions](#podmigrationjob-support-eviction-conditions)
+	- [Table of Contents](#table-of-contents)
+	- [Summary](#summary)
+	- [Motivation](#motivation)
+		- [Goals](#goals)
+		- [Non-Goals/Future Work](#non-goalsfuture-work)
+	- [Proposal](#proposal)
+		- [User Stories](#user-stories)
+			- [Story 1](#story-1)
+			- [Story 2](#story-2)
+		- [API Changes](#api-changes)
+		- [Configuration API](#configuration-api)
+		- [Workflow](#workflow)
+			- [Basic workflow illustration](#basic-workflow-illustration)
+			- [Notes](#notes)
+
+
+## Summary
+
+PodMigrationJob supports two types of pod migration modes currently, i.e. `EvictDirectly` and `ReservationFirst`. Pod eviction is performed directly, or after Reservation is successfully scheduled. However, as an operation at high risk, evicting pod must meet some external interception conditions in scenarios with high technical risk requirements (such as financial scenarios). Typical eviction conditions include cutting off North/South & East/West traffic, getting approval from change risk control system, etc. Therefore, the eviction conditions mechanism is needed to let the PodMigrationJob check whether the predefined interception conditions have been met before performing the actual eviction operation, reducing the risk of pod migration.
+
+## Motivation
+
+### Goals
+
+- Introduce mechanism for PodMigrationJob to make sure that some eviction conditions are met before performing actual evictions on pods.
+
+### Non-Goals/Future Work
+
+- Embed validation logics of various eviction conditions inside PodMigrationJob controller.
+- Illustrate details of eviction conditions.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+Some users have built their own operators to fully manage the North/South & East/West traffic during the whole lifecycle of their pods, so that errors and exceptions could be avoided when containers are created or destroyed. The mission of traffic operator can be simply summarized as automatically turning on traffic before the pod is brought online, and automatically turning off traffic before the pod goes offline. Through the PodMigrationJob eviction condition mechanism, the condition that traffic is turned off before evicting the pod can be ensured. This helps users to reduce technological risks during the pod migration process. 
```

Comment:
Yeah, sounds similar but not the same. Traffic operator is able to avoid traffic loss in finer-grained way by extending the native lifecycle of pods as described in [PodOpsLifecycle](https://www.kusionstack.io/docs/operating/concepts/podopslifecycle) .